tests/llm/test_backends/test_openai.py

```
@@ -1210,3 +1216,815 @@ async def test_openai_backend_structured_without_tools_still_uses_parse() -> Non
+                    "item": {
+                        "type": "message",
+                        "content": [{"type": "refusal", "refusal": "unsafe request"}],
+                    },
+                }
+            ],
+            final,
+        )
+    )
+
+    chunks: list[StreamChunk] = []
+    with pytest.raises(ValidationException, match="^Responses provider refusal$"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)